torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -646,6 +722,7 @@ def triton_kernel_wrapper_mutation_functionalize(
             kernel_idx=kernel_idx,
             constant_args_idx=constant_args_idx,
             grid=grid,
+            tma_descriptor_metadata=tma_descriptor_metadata,
```

Comment:
it's possible to mutate the tensor that that the TMA descriptor points to, right? We need to functionalize that if it happens.

Reply:
From offline discussion, by "mutate the tensor that that the TMA descriptor points to" you're referring to mutating via the TMA descriptor which can only happen in a Triton kernel. This should be covered, as I've added `tt.experimental_descriptor_store` as a new mutating TTIR op to the Triton kernel analysis pass. At the AOTAutograd tracing time, the arguments to the Triton kernels are the actual tensors underlying the TMA descriptors. As a result, the analysis pass will mark the mutated underlying tensors and they will be added to `tensors_to_clone`. Does this solve the issue you've raised?